tabbycat/users/models.py

```diff
@@ -0,0 +1,46 @@
+from django.conf import settings
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from utils.fields import ChoiceArrayField
+
+from .permissions import Permission
+
+
+class UserPermission(models.Model):
```

Comment:
Should there be some hook here that expires the cached permissions upon save? I think the caching might be interfering with when a permission is revoked? Or more precisely it wouldn't take effect until the cache next expires, would that be okay with tab directors?